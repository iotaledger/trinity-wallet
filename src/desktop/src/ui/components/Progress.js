import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import css from './progress.scss';

/**
 * Progress bar component
 */
export default class Progress extends React.PureComponent {
    static propTypes = {
        /** Progress bar progress */
        progress: PropTypes.number.isRequired,
        /** Progress bar title */
        title: PropTypes.string,
        /** Progress bar subtitle */
        subtitle: PropTypes.string,
        /** Progress bar style type */
        type: PropTypes.oneOf(['large']),
    };

    render() {
        const { progress, title, subtitle, type } = this.props;

        return (
            <div className={classNames(css.progress, type ? css[type] : null)}>
                {title && <p>{title}</p>}
                <div>
                    <div className={css.bar} style={{ width: `${Math.min(progress, 100)}%` }} />
                </div>
                {subtitle && <small>{subtitle}</small>}
            </div>
        );
    }
}
