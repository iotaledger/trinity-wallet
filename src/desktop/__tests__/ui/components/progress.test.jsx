import React from 'react';
import { shallow } from 'enzyme';

import Progress from 'ui/components/Progress';

const props = {
    progress: 45,
    title: 'Dolor sit amet',
    subtitle: 'Lorem ipsum',
    type: 'large',
};

describe('Progress component', () => {
    test('Render the component', () => {
        const wrapper = shallow(<Progress {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Correct progress', () => {
        const wrapper = shallow(<Progress {...props} />);

        expect(
            wrapper
                .find('.bar')
                .first()
                .props().style.width,
        ).toEqual(`${props.progress}%`);
    });

    test('With title', () => {
        const wrapper = shallow(<Progress {...props} />);
        const titleEl = wrapper.find('p');

        expect(titleEl).toHaveLength(1);
        expect(titleEl.text()).toEqual(props.title);
    });

    test('Without title', () => {
        const mockProps = Object.assign({}, props, { title: null });
        const wrapper = shallow(<Progress {...mockProps} />);

        expect(wrapper.find('p')).toHaveLength(0);
    });

    test('With subtitle', () => {
        const wrapper = shallow(<Progress {...props} />);
        const titleEl = wrapper.find('small');

        expect(titleEl).toHaveLength(1);
        expect(titleEl.text()).toEqual(props.subtitle);
    });

    test('Without subtitle', () => {
        const mockProps = Object.assign({}, props, { subtitle: null });
        const wrapper = shallow(<Progress {...mockProps} />);

        expect(wrapper.find('small')).toHaveLength(0);
    });
});
