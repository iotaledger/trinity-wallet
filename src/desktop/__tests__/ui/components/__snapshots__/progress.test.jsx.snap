// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress component Render the component 1`] = `
<div
  className="progress large"
>
  <p>
    Dolor sit amet
  </p>
  <div>
    <div
      className="bar"
      style={
        Object {
          "width": "45%",
        }
      }
    />
  </div>
  <small>
    Lorem ipsum
  </small>
</div>
`;
